### DISCUSSION TITLE
Disable entry in lookup editor

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5822] [Help]
I have same lookup editor hooked to multiple fields and I want to disable entry in lookupeditor If already selected before.
Eg.
LookupEditor field - Property Name (P1, P2, P3, P4)

Fields to which Property Lookupeditor is attached
Field 1
Field 2
Field 3

If Property P1 is selected for Field 1
Than Only P2, P3, P4 will be available for selection whereas P1 will be readily or disabled in lookup editor

**Comment by minhhungit** [comment]
using ServiceLookup is a good solution

**Reply by TechAspirant** [reply]
Any sample ?

**Reply by minhhungit** [reply]
another way I guess you can inherit a new lookup editor from Serenity.LookupEditorBase and custom select2 option 
Just search by keyword `LookupEditorBase` or `Lookup service` on github, I think you can find examples by searching

![image](https://user-images.githubusercontent.com/2279508/123329336-e3ca5280-d566-11eb-9222-6817aa52c0ac.png)

**Reply by minhhungit** [reply]
Try this, fast coding , I have not tested it

![image](https://user-images.githubusercontent.com/2279508/123330095-c6e24f00-d567-11eb-854c-2d549fa9563e.png)

**Reply by minhhungit** [reply]
I checked, get_items does not work, use updateItems, it should work

![image](https://user-images.githubusercontent.com/2279508/123335515-c5685500-d56e-11eb-8cfa-5f8792f01330.png)

**Reply by TechAspirant** [reply]
I tried it disables alternate entry. How to check that the entry is selected or mapped to field already and If selected make it disable ?

**Reply by minhhungit** [reply]
I checked it more careful, I see that we have a method called `getItemDisabled` in `Serenity.LookupEditorBase`, just overwrite it

```javascript
@Serenity.Decorators.registerEditor()
export class TestLookup extends Serenity.LookupEditorBase<Serenity.LookupEditorOptions, StateRow>{

    constructor(container: JQuery, opt: Serenity.LookupEditorOptions) {
        super(container, opt);
    }

    protected getItemDisabled(item: StateRow, lookup: Q.Lookup<StateRow>) {
        if (item.Id % 2 == 0) {
            return true;
        }

        return false;
    }   
}
```

Use `item` and `lookup` param in the method to determind which record you want to hide, to get custom lookup items then search in github with keyword `lookup` please, I also anwsered many times
